src/plugin/BodyROSItem.cpp

```diff
@@ -455,7 +455,61 @@ void BodyROSItem::updateVisionSensor
     vision.step = sensor->image().width() * sensor->image().numComponents();
     vision.data.resize(vision.step * vision.height);
     std::memcpy(&(vision.data[0]), &(sensor->image().pixels()[0]), vision.step * vision.height);
-    publisher.publish(vision);
+
+    sensor_msgs::CameraInfo info;
+    info.header.stamp = vision.header.stamp;
+    info.header.frame_id  = vision.header.frame_id ;
+    info.width = vision.width;
+    info.height = vision.height;
+    info.distortion_model = "plumb_bob";
+    info.D.resize(5);
+    for (int i = 0; i < 5; ++i) {
+        info.D[i] = 0;
+    }
```

Comment:
You can initialize it like this: 
```suggestion
    info.D.resize(5, 0.0);
```